prompting/rewards/date.py

```diff
@@ -11,38 +15,54 @@ def name(self) -> str:
 
     def __init__(self, **kwargs):
         super().__init__()
+    
+    def date_diff(self, ref_date, comp_date):
+        """
+        Calculates the absolute difference in days between two dates.
+        """
+        if not ref_date or not comp_date:
+            return 300
```

Comment:
why?

Reply:
If there is an error in parsing the dates, because we know the format of the reference, the completions date is invalid, therefore a large int is returned to gurantee a 0. Definitely wonky and has been changed to be more clear.